channeldb/channel.go

```diff
@@ -2670,15 +2749,29 @@ func (c *OpenChannel) AdvanceCommitChainTail(fwdPkg *FwdPkg,
 		newRemoteCommit = nil
 	})
 	if err != nil {
-		return err
+		return nil, err
 	}
 
 	// With the db transaction complete, we'll swap over the in-memory
 	// pointer of the new remote commitment, which was previously the tip
 	// of the commit chain.
 	c.RemoteCommitment = *newRemoteCommit
 
-	return nil
+	return finalHtlcs, nil
+}
+
+func putFinalHtlc(finalHtlcsBucket kvdb.RwBucket, id uint64,
+	settled bool) error {
+
+	var key [8]byte
+	byteOrder.PutUint64(key[:], id)
+
+	var value [1]byte
+	if settled {
+		value[0] = 1
+	}
+
+	return finalHtlcsBucket.Put(key[:], value[:])
```

Comment:
>This should take up less space than the revocation log since that uses >40 bytes for each htlc, whereas this uses about 17 and only when it's getting resolved.

@Crypt-iQ, how do you get to 17?

Reply:
Ah right. In the PR as it is right now, there is a bucket per channel. So the shortchanid isn't part of the key, making I think for a total of 9 bytes.

Reply:
Oh right even less